subsys/logging/log_backend_rpc_history_ram.c

```diff
@@ -28,23 +28,32 @@ void log_rpc_history_init(void)
 
 void log_rpc_history_push(const union log_msg_generic *msg)
 {
-	uint32_t wlen;
-	union log_msg_generic *copy;
-	int len;
+	size_t wlen;
+	union mpsc_pbuf_generic *dst;
+	uint8_t *dst_data;
+	uint8_t *src_data;
+
+	wlen = log_msg_generic_get_wlen(msg);
+	dst = mpsc_pbuf_alloc(&log_history_pbuf, wlen, K_NO_WAIT);
+	if (!dst) {
+		/* No space to store the log */
+		return false;
+	}
 
-	wlen = log_msg_generic_get_wlen(&msg->buf);
-	copy = (union log_msg_generic *)mpsc_pbuf_alloc(&log_history_pbuf, wlen, K_NO_WAIT);
+	/* First word contains internal mpsc packet flags and when copying
+	 * those flags must be omitted.
+	 */
+	dst_data = (uint8_t *)dst + sizeof(struct mpsc_pbuf_hdr);
+	src_data = (uint8_t *)msg + sizeof(struct mpsc_pbuf_hdr);
 
-	if (!copy) {
-		return;
+	if (wlen * sizeof(uint32_t) <= sizeof(struct mpsc_pbuf_hdr)) {
```

Comment:
This should probably be an assert, or should be checked before `mpsc_pbuf_alloc`. Once `mpsc_pbuf_alloc` is called, we should always call `mpsc_pbuf_commit` and not return earlier.

Reply:
Good point, moved it before the allocation